tests/unit/_modules/test_cephprocesses.py

```diff
@@ -0,0 +1,506 @@
+    def test_check_osd(self, insuf_mock, filter_mock, mc):
+        """
+        Filter for returns something
+        insuff is True
+        """
+        mc.running = True
+        filter_mock.return_value = ['notnull']
+        mc.insufficient_osd_count = True
+        mc.check_osds()
+        assert mc.running == False
+
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.filter_for')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck._insufficient_osd_count')
+    def test_check_osd_1(self, insuf_mock, filter_mock, mc):
+        """
+        filter_for does not return
+        """
+        mc.running = True
+        filter_mock.return_value = []
+        mc.check_osds()
+        assert mc.running == True
+
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.filter_for')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck._insufficient_osd_count')
+    def test_check_osd_2(self, insuf_mock, filter_mock, mc):
+        """
+        filter_for does return something
+        but isuff is False
+        """
+        mc.running = True
+        filter_mock.return_value = ['notnull']
+        mc.insufficient_osd_count = False
+        mc.check_osds()
+        assert mc.running == True
+
+    def test_report_empty(self, mc):
+        mc.up = []
+        mc.down = []
+        assert mc.report() == {'up': {}, 'down': {}}
+
+    def test_report_up(self, mc):
+        role_name = 'mds'
+        proc_name = 'ceph-mds'
+        proc = self.build_proc(role_name, proc_name)
+        mc.up = [proc]
+        mc.down = []
+        assert mc.report() == {'up': {'ceph-mds': [0]}, 'down': {}}
+
+    def test_report_down(self, mc):
+        role_name = 'mds'
+        proc_name = 'ceph-mds'
+        mc.up = []
+        mc.down = [proc_name]
+        assert mc.report() == {'up': {}, 'down': {'ceph-mds': 'ceph-mds'}}
+
+    def test_report_up_down(self, mc):
+        role_name = 'storage'
+        proc_name = 'ceph-osd'
+        proc_name_down = 'ceph-mds'
+        proc = self.build_proc(role_name, proc_name)
+        mc.up = [proc]
+        mc.down = [proc_name_down]
+        assert mc.report() == {'up': {'ceph-osd': [0]}, 'down': {'ceph-mds': 'ceph-mds'}}
+
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck._missing_osds', new_callable=mock.PropertyMock)
+    def test_report_up_down_down(self, missing_mock, mc):
+        """
+        expected 1,3 up
+        0 is down
+        """
+        # 2 is blacklisted
+        missing_mock.return_value = ['1', '3']
+        role_name = 'storage'
+        proc_name = 'ceph-osd'
+        proc_name_down = 'ceph-mds'
+        mc.insufficient_osd_count = True
+        proc = self.build_proc(role_name, proc_name)
+        mc.up = [proc]
+        mc.down = [proc_name_down]
+        assert mc.report() == {'up': {'ceph-osd': [0]}, 'down': {'ceph-mds': 'ceph-mds', 'ceph-osd': ['1', '3']}}
+
+
+class TestInstanceMeth():
```

Comment:
Meth is Methods i assume?
I would use the Methods :)

Reply:
lazy me :)